dom-fetch-client/src/main/scala/org/http4s/dom/FetchReferrer.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2021 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.dom
+
+import org.http4s.Uri
+import org.http4s.util.Writer
+import org.http4s.util.Renderable
+
+sealed trait FetchReferrer extends Renderable
+object FetchReferrer {
+  final case object NoReferrer extends FetchReferrer {
+    override def render(writer: Writer): writer.type = writer << "no-referrer"
+  }
+  final case object Client extends FetchReferrer {
+    override def render(writer: Writer): writer.type = writer << "client"
+  }
+  final case class URL(uri: Uri) extends FetchReferrer {
```

Comment:
Shouldn't this be just a path, since the docs say it must be a same origin URL?

> referrer
>   A USVString specifying the referrer of the request. This can be a same-origin URL, about:client, or an empty string.

@armanbilge do we have access to the host when we create the `Fetch` client? I'm still coming up to speed on Scalajs, but if we have the host then we can use that to set the `referrer` to `<HOST>/<PATH>`. If we take in a full `Uri` it will be ignored if it is not the same origin, which is probably not ideal.

Reply:
@isomarcte yes, in the browser we do have access to `window.location.host` for the window and also `self.location.host` for web workers. For `FetchClient` to work in both of these browser environments we'll have to do this carefully 🤔 

But 👍 on using a path here instead, it narrows statically to only valid inputs.

Reply:
I definitely agree on this.

It may be tricky to support it correctly on all platforms though. As @armanbilge mentioned, we should proceed differently whether we are in window or web worker. Also, do we want to support `node-fetch`? (I have no idea what `node-fetch` does with the `referrer` option).